lib/rubyfit/helpers.rb

```diff
@@ -0,0 +1,107 @@
+module RubyFit::Helpers
+  # Garmin timestamps start at 12:00:00 01-01-1989, 20 years after the unix epoch
+  GARMIN_TIME_OFFSET = 631065600 
+
+  DEGREES_TO_SEMICIRCLES = 2**31 / 180.0
+
+  # Converts a fixnum or bignum into a byte array, optionally
+  # truncating or right-filling with 0 to match a certain size
+  def num2bytes(num, byte_count, big_endian = true)
+    raise ArgumentError.new("num must be an integer") unless num.is_a?(Integer)
+    orig_num = num
+    # Convert negative numbers to two's complement (1-byte alignment)
+    if num < 0
+      num = num.abs
+
+      if num > 2 ** (byte_count * 8 - 1)
+        STDERR.puts("RubyFit WARNING: Integer underflow for #{orig_num} (#{orig_num.bit_length + 1} bits) when fitting in #{byte_count} bytes (#{byte_count * 8} bits)")
+      end
+
+      num = 2 ** (byte_count * 8) - num
```

Comment:
This is kinda a nitpick, but, you could store part of this calculation and re-use the expensive part of it for lines 16 and 20. I only say it because num2bytes is called a ton when writing anything. *shrug*